fulp_modules/devils/devil.dm

```diff
@@ -0,0 +1,382 @@
+		return FALSE
+	if(be_close && !in_range(M, src))
+		to_chat(src, "<span class='warning'>You are too far away!</span>")
+		return FALSE
+	return TRUE
+
+/// Beepsky doesn't like Devils
+/mob/living/carbon/true_devil/assess_threat(judgement_criteria, lasercolor = "", datum/callback/weaponcheck=null)
+	return 666
+
+/mob/living/carbon/true_devil/soundbang_act()
+	return FALSE
+
+/mob/living/carbon/true_devil/get_ear_protection()
+	return 2
+
+/mob/living/carbon/true_devil/attacked_by(obj/item/I, mob/living/user, def_zone)
+	apply_damage(I.force, I.damtype, def_zone)
+	var/message_verb = ""
+	if(length(I.attack_verb_continuous))
+		message_verb = "[pick(I.attack_verb_continuous)]"
+	else if(I.force)
+		message_verb = "attacked"
+
+	var/attack_message = "[src] has been [message_verb] with [I]."
+	if(user)
+		user.do_attack_animation(src)
+		if(user in viewers(src, null))
+			attack_message = "[user] has [message_verb] [src] with [I]!"
+	if(message_verb)
+		visible_message("<span class='danger'>[attack_message]</span>",
+		"<span class='userdanger'>[attack_message]</span>", null, COMBAT_MESSAGE_RANGE)
+	return TRUE
+
+/mob/living/carbon/true_devil/can_be_revived()
+	return TRUE
+
+/// They're immune to fire.
+/mob/living/carbon/true_devil/resist()
+	set name = "Resist"
+	set category = "IC"
+
+	if(!can_resist())
+		return
+	changeNext_move(CLICK_CD_RESIST)
+
+	SEND_SIGNAL(src, COMSIG_LIVING_RESIST, src)
+	if(!HAS_TRAIT(src, TRAIT_RESTRAINED) && pulledby) // Resisting Grabs
+		log_combat(src, pulledby, "resisted grab")
+		resist_grab()
+		return
+
+	if(on_fire)
+		resist_fire() // Resisting out of Fires
+
+/// Intents
+/mob/living/carbon/true_devil/attack_hand(mob/living/carbon/human/M)
+	. = ..()
+	if(.)
+		switch(M.a_intent)
+			if("harm")
+				var/damage = rand(1, 5)
+				playsound(loc, "punch", 25, TRUE, -1)
+				visible_message("<span class='danger'>[M] punches [src]!</span>", \
+						"<span class='userdanger'>[M] punches you!</span>")
+				adjustBruteLoss(damage)
+				log_combat(M, src, "attacked")
+				updatehealth()
+			if("disarm")
+				if(!(mobility_flags & MOBILITY_STAND) && !ascended) //No stealing the arch devil's pitchfork.
+					if(prob(5))
+						Unconscious(40)
+						playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
+						log_combat(M, src, "pushed")
+						visible_message("<span class='danger'>[M] pushes [src] down!</span>", \
+							"<span class='userdanger'>[M] pushes you down!</span>")
+					else
+						if(prob(25))
+							dropItemToGround(get_active_held_item())
+							playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
+							visible_message("<span class='danger'>[M] disarms [src]!</span>", \
+							"<span class='userdanger'>[M] disarms you!</span>")
+						else
+							playsound(loc, 'sound/weapons/punchmiss.ogg', 25, TRUE, -1)
+							visible_message("<span class='danger'>[M] fails to disarm [src]!</span>", \
+							"<span class='userdanger'>[M] fails to disarm you!</span>")
```

Comment:
holy shit, this is reminding me of monkey code

Reply:
probably because devils are monkey brains.